sa/storage-authority.go

```diff
@@ -429,7 +472,30 @@ func (ssa *SQLStorageAuthority) UpdatePendingAuthorization(authz core.Authorizat
 		return
 	}
 
+	var challs []challengeModel
+	_, err = tx.Select(&challs, "SELECT * FROM challenges WHERE authorizationID = :authID", map[string]interface{}{"authID": authz.ID})
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+	if len(challs) != len(authz.Challenges) {
+		return fmt.Errorf("Invalid number of challenges provided")
+	}
+	for i, authChall := range authz.Challenges {
+		chall := challengeModel{}
+		chall.Challenge = authChall
+		chall.ID = challs[i].ID
+		chall.AuthorizationID = challs[i].AuthorizationID
+		_, err = tx.Update(&chall)
+		if err != nil {
+			tx.Rollback()
+			return err
+		}
+	}
+
 	err = tx.Commit()
+	if err != nil {
```

Comment:
It looks like you missed handling this error? Or maybe the idea is that you'd just return the error anyhow? In which case, just skip the if statement.